src/app/UserService.ts

```diff
@@ -1,77 +1,69 @@
-import { DynamoDB } from "aws-sdk";
+import { DynamoDocumentClient } from "src/shared/DynamoDocumentClient";
 import { User } from "./User";
 
 export class UserService {
   private _userTableName: string;
-  private _dbClient: DynamoDB.DocumentClient;
+  private _dbClient: DynamoDocumentClient;
 
   constructor({
     userTableName,
     dbClient,
   }: {
     userTableName: string;
-    dbClient: DynamoDB.DocumentClient;
+    dbClient: DynamoDocumentClient;
   }) {
     this._userTableName = userTableName;
     this._dbClient = dbClient;
   }
 
   get: (email: string) => Promise<User | null> = async (email: string) => {
     const key = { email };
-    const result = await this._dbClient
-      .get({
-        TableName: this._userTableName,
-        Key: key,
-      })
-      .promise();
 
-    const item = result.Item || key;
-    return this.mapUser(item);
+    const result = await this._dbClient.get({
+      TableName: this._userTableName,
+      Key: key,
+    });
+
+    const item = result.Item || { email };
+    return this.mapUser(item as Record<string, unknown>);
```

Comment:
I'm not sure this is the last thing we got to in the pair programming session.

Reply:
Mmm, as I understand, the cast should not be necessary.

Reply:
👍